### DISCUSSION TITLE
In Incident Type a Text or an Array[Text]

### DISCUSSION THREAD

**Discussion by adamjack** [ulfsri/neris-framework#14] [Q&A]
I understand that Incident Types are hierarchical with up to three components, and I understand that there can be multiple final types. That said, I believe I see two ways to represent this in the schema, and suspect there should be only one way.

In the dispatch module it appears to mean "dispatched to one thing" but it is "Text" whereas in the incident module where there can be N types it is Array[Array[Text]]. In the dispatch module I suspect the || separator within a string serves the purpose of one level of Array[Text].

e.g.

   "incident_types": [
        "FIRE||OUTSIDE_FIRE||CONSTRUCTION_WASTE"
     ],

So does NERIS want incident types as || separated strings or arrays of string?

regards,

Adam

**Comment by cweinschenk** [comment]
Hi Adam,

For incident type, the payload should come in a double pipe separated strings. We pushed an updated payload example in 0.1.37 that should hopefully make things more clear: https://api-test.neris.fsri.org/v1/docs#/Incidents/create_incident_incident__neris_id_entity__post

For dispatch it is a single versus multiple by nature of dispatch. On the dispatch side, what we are after is the local dispatch code (incident_type) so that a department can still leverage that information for their local analytics, while also building a map/dictionary that connects the local codes to the NERIS incident type to better allow cross department analysis.

**Comment by adamjack** [comment]
Thanks Craig,

This question came from the YML files (e.g. next block), but I'm pretty much moving to the OpenAPI specification now, so no big deal.

  incident_final_type:
  group: type_identifiers
  type: Array[Array[Text]]
  value_set: 'TRUE'
  format: See value_1, value2, value_3 from type_incident

Thanks

Adam
